### ISSUE TITLE
[Feature request] Better sliders

### ISSUE THREAD

**Issue by Lucaspec72** [Mr-Wiseguy/Zelda64Recomp#192]
Currently the sliders just move by one point when holding left or right, which means you have to repeatedly mash your buttons to change their values. It'd be nice to just hold the dpad or joystick for it to move across the range of the slider.

**Comment by briaguya0** [comment]
this will likely require reworking how controller input is sent to RmlUi

https://github.com/Zelda64Recomp/Zelda64Recomp/blob/6121e5aa3c3c01fb0011d1bb5edd2d33061de178/src/ui/ui_renderer.cpp#L1181-L1214

relies on [`SDL_ControllerButtonEvent`](https://wiki.libsdl.org/SDL2/SDL_ControllerButtonEvent) which is fired when a button is pressed or released, but not when held

> ```c
> typedef struct SDL_ControllerButtonEvent
> {
>    Uint32 type;        /**< SDL_CONTROLLERBUTTONDOWN or SDL_CONTROLLERBUTTONUP */
>  ```

**Comment by flamespeedy** [comment]
with mouse i can change things quickly, with controller it's a nightmare

**Comment by dcvz** [comment]
This has now been implemented by #364 and will be available in the next release!

**Comment by briaguya0** [comment]
not quite, this issue is still present when navigating the menu using a game controller (for the reasons described in my previous comment) - game controller button down events don't have the repeat functionality keypress events do

**Comment by dcvz** [comment]
> not quite, this issue is still present when navigating the menu using a game controller (for the reasons described in my previous comment) - game controller button down events don't have the repeat functionality keypress events do

Seems like I was testing with Steam input turned on, where that does indeed work. Well still have to fix this for game controllers.

**Comment by Reonu** [comment]
This was fixed in the 1.2.0 update

**Comment by briaguya0** [comment]
seems to specifically have been fixed by https://github.com/Zelda64Recomp/Zelda64Recomp/commit/3cbc871194b095d1c85f53bffc141b1ab2011ff8
